File: windows/tests/ConnectorControl.Core.Tests/TestSupport/FakeNotifier.cs
using ConnectorControl.Core.Services;

namespace ConnectorControl.Core.Tests.TestSupport;

public sealed class FakeNotifier : INotifier
{
    public List<(string Title, string Body, string? Category)> Sent { get; } = [];

    public event Action? RestartActionActivated;

    public void Notify(string title, string body, string? category = null) => Sent.Add((title, body, category));

    /// <summary>Simulates the user clicking the toast's Restart Claude button.</summary>
    public void ActivateRestart() => RestartActionActivated?.Invoke();
}

File: Tests/ConnectorControlStateTests/TestSupport/FakeNotifier.swift
@testable import ConnectorControlState

final class FakeNotifier: Notifier {
    struct Sent: Equatable {
        let title: String
        let body: String
        let category: String?
    }

    private(set) var sent: [Sent] = []
    var onRestartAction: MainActorAction?

    func notify(title: String, body: String, category: String?) {
        sent.append(Sent(title: title, body: body, category: category))
    }

    func clearSent() { sent.removeAll() }

    /// Simulates the user clicking the notification's Restart Claude button.
    @MainActor
    func activateRestart() { onRestartAction?() }
}
